libs/utils/src/main/java/com/akto/log/CacheLoggerMaker.java

```diff
@@ -0,0 +1,60 @@
+package com.akto.log;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public class CacheLoggerMaker extends LoggerMaker {
+
+    private static final int CACHE_LIMIT = 50;
+    private static HashMap<String, Integer> errorLogCache = new HashMap<>();
+
+    static {
+        scheduler.scheduleAtFixedRate(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    clearErrorCache();
+                } catch (Exception e) {
+                    internalLogger.error("ERROR: In clearing log cache: " + e.toString());
+                }
+            }
+        }, 2, 5, TimeUnit.MINUTES);
+    }
+
+    public CacheLoggerMaker(Class<?> c) {
+        super(c);
+    }
+
+    public static void clearErrorCache() {
+
+        for (Map.Entry<String, Integer> err : errorLogCache.entrySet()) {
+            String errorMessage = err.getKey() + " : logged " + err.getValue() + " times";
+            sendToSlack(errorMessage);
+        }
+
+        errorLogCache.clear();
+    }
+
+    @Override
+    public void errorAndAddToDb(String err, LogDb db) {
+        basicError(err, db);
+
+        if (db.equals(LogDb.BILLING) || db.equals(LogDb.DASHBOARD)) {
+            try {
+                errorLogCache.put(err, errorLogCache.getOrDefault(err, 0) + 1);
+            } catch (Exception e) {
+                internalLogger.error("ERROR: In adding to log cache: " + e.toString());
+            }
+        }
+
+        if (errorLogCache.size() > CACHE_LIMIT) {
```

Comment:
along with the size based condition, we should also add a time based condition. We will flush logs when either size breaches the CACHE_LIMIT or time condition is reached. We can keep time limit as 1 minute for now.

Reply:
Without this logs might delayed till we reach limit of 50